### ISSUE TITLE
Remove Kube object captures other than latest, complete one, and a partial one in progress

### ISSUE THREAD

**Issue by hatfieldbrian** [RamenDR/ShioRamen#41]
## Background
Backups are created regularly, each successive backup with a unique, monotonically increasing number.  Only a latest, complete backup is need for disaster recovery, which is identified by its number, stored in VRG status, and uploaded to an S3 store for a recovery cluster to access.  There should be sufficient space in an S3 store for 2 backups: the latest complete one, and the previous latest complete one.

## Problem
Backup space is limited.  If there is insufficient space in an S3 store for a backup to be created, then recovery points cease being created and the latest recovery point begins to age such that a recovery point objective, RPO, may no longer be achieved.

## Solution objective
Delete any backup with:
- A number less than the latest complete one
- A number greater than the latest complete one that is not progressing as identified by a failed status

## Lower numbered backup deletion point:
Once a new backup is created, if it is not the first one that is numbered 0, then the previous complete one, typically numbered one less, ought deleted before the next backup starts.  If backup deletion is performed on the same reconcile as a successful backup, then:
- the number of the previous, latest, complete backup is known and need not be discovered
- the next backup has not yet started, because that happens on a subsequent reconcile which may start only after the current one finishes

For MVP, I think deletion at this point is sufficient.  Discovering and deleting other, lower numbered backups may be considered an error case and deferred.

## Higher numbered backup deletion:
There should only be one higher-numbered backup on the local cluster, and it should be one greater than the last protected.  If the ramen or velero operator were to crash, it should be restarted and continue with the current backup.

At recovery time, there should be at most be only one higher-numbered backup, if it exists, it should be deleted.

## Backup deletion methods:
1. Use object storage type agnostic Velero interface: `DeleteBackupRequest` via the Velero driver (`controllers/kubobjects.go`)
1. Use Amazon S3 interface (object storage type specific)
1. Use a Velero backup Kube object and adjust its TTL to zero (not sure if this works if Velero Kube object needs to be recreated)

Velero creates backups.  Though not the only one capable, it makes sense to me that Velero delete backups.  I suggest the 1st method be used.
